hikari/internal/data_binding.py

```diff
@@ -392,3 +394,33 @@ def put_snowflake_array(
         """  # noqa: E501 - Line too long
         if values is not undefined.UNDEFINED:
             self[key] = [str(int(value)) for value in values]
+
+
+def cast_enum_array(cast: typing.Callable[[T_co], T], raw_values: typing.Iterable[T_co], /) -> typing.List[T]:
```

Comment:
Personally, i find `cast_enum_array` a confusing name. Would `cast_variants_array` be better?